integration/airflow/openlineage/airflow/extractors/manager.py

```diff
@@ -0,0 +1,56 @@
+import logging
+from typing import Optional
+
+from openlineage.airflow.extractors import TaskMetadata, BaseExtractor, Extractors
+from openlineage.airflow.utils import get_job_name
+
+
+class ExtractorManager:
+    """Class abstracting management of custom extractors."""
+    def __init__(self):
+        self.extractors = {}
+        self.extractor_mapper = Extractors()
+        self.log = logging.getLogger()
+
+    def extract_metadata(self, dagrun, task, complete: bool, task_instance=None) -> TaskMetadata:
```

Comment:
_Minor_: Would it be sensible to default complete to `False`?

Reply:
Done.